packages/messaging/src/extension.ts

```
@@ -51,9 +60,25 @@ export function defineExtensionMessaging<
 >(config?: ExtensionMessagingConfig): ExtensionMessenger<TProtocolMap> {
   return defineGenericMessanging({
     ...config,
-    sendMessage(message, tabId) {
-      if (tabId == null) return Browser.runtime.sendMessage(message);
-      return Browser.tabs.sendMessage(tabId, message);
+    sendMessage(message, arg) {
+      // No args - send to background
+      if (arg == null) {
+        return Browser.runtime.sendMessage(message);
+      }
+
+      // Handle both number and options object
+      const options: SendMessageOptions = typeof arg === 'number' ? { tabId: arg } : arg;
+
+      if (typeof options.tabId !== 'number') {
+        throw new Error('tabId is required when sending a message to a tab');
+      }
```

I'd rather just let the browser throw an error if the tab ID is missing or the wrong type.

```suggestion
```

---

Fair enough. Done.